docs/development/testing.md

```diff
@@ -437,6 +437,68 @@ gsutil cp -r gs://gardener-prow/pr-logs/pull/gardener_gardener/6136/pull-gardene
   - generally, try to use the default settings for `Eventually` specified via the environment variables
   - only set higher timeouts if waiting for long-running reconciliations to be finished
 
+## Gardener Upgrade Tests (using provider-local)
+Gardener upgrade tests will setup kind cluster and deploys gardener version `vX.X.X` then it upgrades to given version `vY.Y.Y`.
+
+This allow us to qualify/verify the current g/g revision/branch (not yet released) or a specific release whether it is compatible with latest or a specific release (**`GARDENER_PREVIOUS_RELEASE`**).
+
+This helps us to understand real time scenario what happens or how it reacts when gardener upgrades from version `vX.X.X` to `vY.Y.Y` for existing shoots in different states (`creation`/`hibernation`/`wakeup`/`deletion`). Gardener upgrade tests also help us to qualify releases for all flavors **non-HA**, **HA** with failure tolerance `node`/`zone`. 
+
+Just like E2E test also a [KinD cluster](https://kind.sigs.k8s.io/) and [skaffold](https://skaffold.dev/) to bootstrap a full installation of Gardener based on the current revision/branch, including [provider-local](../extensions/provider-local.md).
+This allows us to run e2e tests in an isolated test environment, fully locally without any infrastructure interaction.
+The tests perform a set of operations on Shoot clusters, e.g. create, delete, hibernate and wake up.
+
+Below is a sequence describing how the tests are performed.
+
+- Create a `kind` cluster.
+- Install Gardener version `vX.X.X`.
+- Run gardener pre-upgrade tests which are labeled with `pre-upgrade`.
+- Upgrade Gardener version from `vX.X.X` to `vY.Y.Y`.
+- Run gardener post-upgrade tests which are labeled with `post-upgrade`
+- Tear down seed and cluster.
+
+
+### Debugging e2e Tests
+
+Gardener upgrade tests can be executed for both non-HA and HA environments. Tests are defined here [upgrade.go](../../test/e2e/gardener/upgrade/upgrade.go)
+- `make ci-e2e-gardener-upgrade-kind` -- This will setup a non-HA environment and executes gardener upgrade tests.
+- `make ci-e2e-gardener-upgrade-kind-ha-single-zone` -- This will setup a HA environment with failure tolerance `node` and executes gardener upgrade tests.
+- `make ci-e2e-gardener-upgrade-kind-ha-multi-zone` This will setup a HA environment with failure tolerance `node` and executes gardener upgrade tests.
```

Comment:
```suggestion
- `make ci-e2e-gardener-upgrade-kind-ha-multi-zone` This will setup a HA environment with failure tolerance `node` and execute upgrade tests.
```